include/sos/barriers/DualSOSConeStandardBarrier.tpp

```diff
@@ -0,0 +1,82 @@
+// VolEsti (volume computation and sampling library)
+//
+// Copyright (c) 2020 Bento Natura
+//
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#include "DualSOSConeStandardBarrier.h"
+
+
+template <typename IPMDouble>
+Vector<IPMDouble> DualSOSConeStandardBarrier<IPMDouble>::gradient(Vector x) {
+    assert(in_interior(x));
+    Matrix X = Lambda(x);
+    Matrix Z = X.inverse();
+    Vector g(x.rows());
+    for (int i = 0; i < g.rows(); ++i) {
+        Matrix E_i = Matrix::Zero(X.rows(), X.cols());
+        for (int j = 0; j <= i; ++j) {
+            E_i(j, i - j) = 1;
+        }
+        g(i) = -Z.cwiseProduct(E_i).sum();
+    }
+    return g;
+}
+
+template <typename IPMDouble>
+Matrix<IPMDouble> DualSOSConeStandardBarrier<IPMDouble>::hessian(Vector x) {
+    assert(in_interior(x));
+    Matrix X = Lambda(x);
+    Matrix Z = X.inverse();
+    Matrix H = Matrix::Zero(Z.rows(), Z.cols());
+    for (int u = 0; u < H.rows(); ++u) {
+        for (int v = 0; v < H.cols(); ++v) {
+            IPMDouble H_uv = 0;
+            for (int a = 0; a <= u; ++a) {
+                for (int k = 0; k <= v; ++k) {
+                    H_uv += Z(a, u - a) + Z(k, v - k);
+                }
+            }
+        }
+    }
+    return H;
+}
+
+//FIXME: Wrong implementation in in_interior method.
```

Comment:
is this fixed? How does it affect the code?

Reply:
Yes, fixed and forgot to remove the FIXME.